src/VkBootstrap.h

```diff
@@ -419,16 +445,31 @@ class PhysicalDeviceSelector {
 	// Require a physical device that supports a (major, minor) version of vulkan.
 	PhysicalDeviceSelector& set_minimum_version(uint32_t major, uint32_t minor);
 
-	// Require a physical device which supports the features in VkPhysicalDeviceFeatures.
-	PhysicalDeviceSelector& set_required_features(VkPhysicalDeviceFeatures const& features);
-
+	// Require a physical device which supports a specific set of general/extension features.
+#if defined(VK_API_VERSION_1_1)
+    template <typename T>
+    PhysicalDeviceSelector& add_required_extension_features(T const& features) {
```

Comment:
Just realized I don't think there is any logic to handle 'user added a features2 struct containing the VkPhysicalDeviceFeatures they want'. We use only the ones given by set_required_features. Afaik the code drops those features requested. I don't think its worth changing the code over, but possibly adding more info about it in the documentation would be good.
"// if the struct VkPhysicalDeviceFeatures2 is given to this function, the features member will be ignored. Use set_required_features instead."

Reply:
And I only now realize that you  already have an assert here. That's even better. (Yes, I completely glossed over it in reviewing the code).